js/exporter/exceljs/export.js

```diff
@@ -99,29 +99,48 @@ export const Export = {
         }
     },
 
-    tryGetMergeRange: function(rowIndex, cellIndex, mergedCells, dataProvider) {
-        if(!mergedCells[rowIndex] || !mergedCells[rowIndex][cellIndex]) {
-            const cellMerge = dataProvider.getCellMerging(rowIndex, cellIndex);
-            if(cellMerge.colspan || cellMerge.rowspan) {
-                for(let i = rowIndex; i <= rowIndex + cellMerge.rowspan || 0; i++) {
-                    for(let j = cellIndex; j <= cellIndex + cellMerge.colspan || 0; j++) {
-                        if(!mergedCells[i]) {
-                            mergedCells[i] = [];
-                        }
-                        mergedCells[i][j] = true;
-                    }
-                }
+    tryGetMergeRange: function(excelCell, mergedCells, rowIndex, cellIndex, dataProvider, mergeRowFieldValues, mergeColumnFieldValues, privateOptions) {
+        const { rowspan, colspan } = dataProvider.getCellMerging(rowIndex, cellIndex);
+
+        if(colspan || rowspan) {
+            const needMergeRange = privateOptions._isRangeMerged(dataProvider, rowIndex, cellIndex, rowspan, colspan, mergeRowFieldValues, mergeColumnFieldValues);
+            this.updateMergedCells(excelCell, mergedCells, rowIndex, cellIndex, rowspan, colspan, needMergeRange);
+
+            if(needMergeRange) {
                 return {
-                    start: { row: rowIndex, column: cellIndex },
-                    end: { row: rowIndex + (cellMerge.rowspan || 0), column: cellIndex + (cellMerge.colspan || 0) }
+                    masterCell: excelCell,
+                    ...{ rowspan, colspan }
+                };
+            }
+        }
+    },
+
+    cellInMergeRanges(mergedCells, rowIndex, cellIndex) {
+        return mergedCells[rowIndex] && mergedCells[rowIndex][cellIndex];
+    },
+
+    updateMergedCells(excelCell, mergedCells, rowIndex, cellIndex, rowspan, colspan, needMergeRange) {
+        for(let i = rowIndex; i <= rowIndex + rowspan; i++) {
+            for(let j = cellIndex; j <= cellIndex + colspan; j++) {
+                if(!mergedCells[i]) {
+                    mergedCells[i] = [];
+                }
+                mergedCells[i][j] = {
+                    masterCell: excelCell,
+                    merged: needMergeRange
                 };
             }
         }
     },
 
     mergeCells: function(worksheet, topLeftCell, mergeRanges) {
         mergeRanges.forEach((mergeRange) => {
-            worksheet.mergeCells(mergeRange.start.row + topLeftCell.row, mergeRange.start.column + topLeftCell.column, mergeRange.end.row + topLeftCell.row, mergeRange.end.column + topLeftCell.column);
+            const startRowIndex = mergeRange.masterCell.fullAddress.row;
```

Comment:
I seems that our 'mergeRanges' got enough code to be a separate class

Reply:
9c50f59